File: backend/core/src/main/java/ch/qarts/tattool/core/domain/recording/Recording.java
package ch.qarts.tattool.core.domain.recording;

import ch.qarts.tattool.core.domain.command.Payload;
import ch.qarts.tattool.core.domain.dataset.Dataset;
import lombok.*;

import java.util.ArrayList;
import java.util.List;

@AllArgsConstructor
@NoArgsConstructor
@Builder
@Data
@EqualsAndHashCode
public class Recording {

    private String id;
    private String name;
    private String description;
    private String platform;
    private List<String> tags;
    private Dataset dataset;

    private List<RecordingStep<? extends Payload>> recordingStepList = new ArrayList<>();
}

File: frontend/tattool-ts/src/model/transport/Recording.ts
import { Dataset } from "./Dataset";
import { RecordingStep } from "./RecordingStep";

interface Recording {
    readonly id?: string;
    readonly name?: string;
    readonly description?: string;
    readonly platform?: string;
    readonly dataset?: Dataset;
    readonly tags: Array<string>;
    readonly recordingStepList?: Array<RecordingStep>;
}


export { Recording }